src/Parser/ExponentialMoneyParser.php

```diff
@@ -0,0 +1,119 @@
+<?php
+
+namespace Money\Parser;
+
+use Money\Currencies;
+use Money\Currency;
+use Money\Exception\ParserException;
+use Money\Money;
+use Money\MoneyParser;
+use Money\Number;
+
+/**
+ * Parses an exponential string into a Money object.
+ *
+ * @example `2.8865798640254e+15`
+ *
+ * @author George Mponos <gmponos@gmail.com>
+ */
+final class ExponentialMoneyParser implements MoneyParser
+{
+    const EXPO_DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>0|[1-9]\d*)?\.?(?P<fraction>\d+)?[eE][-+]\d+$/';
+
+    const DECIMAL_PATTERN = '/^(?P<sign>-)?(?P<digits>0|[1-9]\d*)?\.?(?P<fraction>\d+)?$/';
+
+    /**
+     * @var Currencies
+     */
+    private $currencies;
+
+    /**
+     * @param Currencies $currencies
+     */
+    public function __construct(Currencies $currencies)
+    {
+        $this->currencies = $currencies;
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function parse($money, $forceCurrency = null)
+    {
+        if (!is_string($money)) {
+            throw new ParserException('Formatted raw money should be string, e.g. 1.00');
+        }
+
+        if (null === $forceCurrency) {
+            throw new ParserException(
+                'DecimalMoneyParser cannot parse currency symbols. Use forceCurrency argument'
```

Comment:
Shouldn't this say "ExponentialMoneyParser", not "DecimalMoneyParser"?

Reply:
Done.. nice catch